docker-compose.yml

```
@@ -1,5 +1,38 @@
 version: '3'
 services:
+
+  backend:
+    image: apachepulsarui/backend
+    container_name: backend
+    profiles:
+      - backend
+    build: ./backend
+    ports:
+      - "8081:8081"
+    environment:
+      - "PULSAR_CONSUMER_URL=pulsar://pulsar:6650"
+      - "PULSAR_ADMIN_URL=http://pulsar:8080"
+    depends_on:
+      pulsar:
+        condition: service_healthy
+
+  frontend:
+    image: apachepulsarui/frontend
+    container_name: frontend
+    profiles:
+      - frontend
+    env_file:
+      - .env
+    build:
+      context: ./frontend
+      args:
+        - "BACKEND_URL=http://${BACKEND_IP}:8081"
```

Could we use `localhost` as a default value for `BACKEND_IP` here? e.g.
```suggestion
        - "BACKEND_URL=http://${BACKEND_IP:-localhost}:8081"
```
Not sure if this works. But if it does, you could remove the part from the readme for local startup.

---

Ah yeah, I've seen Java Apps where this works - I'll try it out later. Thanks for the hint!

---

Does not work as easy because of the `env_file` key - an `.env` file is required if this key is set in the docker-compose.